modules/payment/src/Plugin/Commerce/CheckoutPane/PaymentProcess.php

```diff
@@ -133,38 +133,43 @@ public function buildPaneForm(array $pane_form, FormStateInterface $form_state,
     $payment_gateway = $this->order->payment_gateway->entity;
     $payment_gateway_plugin = $payment_gateway->getPlugin();
 
+    $payment_storage = $this->entityTypeManager->getStorage('commerce_payment');
+    $payment = $payment_storage->create([
```

Comment:
For offsite we are creating but not saving the payment? In general, our idea was to not create the payment until the user comes back, if possible. That way we don't get an entity per redirect.